src/test/scala/io/iohk/ethereum/blockchain/sync/BlockBroadcastSpec.scala

```diff
@@ -23,30 +23,74 @@ class BlockBroadcastSpec extends FlatSpec with Matchers  {
     val secondBlock = NewBlock(Block(secondHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty + 2)
 
     //when
-    blockBroadcast.broadcastNewBlocks(Seq(firstBlock, secondBlock), Map(peer -> initialPeerInfo))
+    blockBroadcast.broadcastBlocks(Seq(firstBlock, secondBlock), Map(peer -> initialPeerInfo))
 
     //then
     etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(secondBlock, peer.id))
+    etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(secondBlockNewHashes, peer.id))
     etcPeerManagerProbe.expectNoMsg()
   }
 
   it should "send a new block only when it is not known by the peer (known by comparing max block number)" in new TestSetup {
     //given
     val firstHeader: BlockHeader = baseBlockHeader.copy(number = initialPeerInfo.maxBlockNumber + 4)
+    val firstBlockNewHashes = NewBlockHashes(Seq(PV62.BlockHash(firstHeader.hash, firstHeader.number)))
     val firstBlock = NewBlock(Block(firstHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty - 2)
 
     //Block should already be known by the peer due to max block known
     val secondHeader: BlockHeader = baseBlockHeader.copy(number = initialPeerInfo.maxBlockNumber - 2)
     val secondBlock = NewBlock(Block(secondHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty - 2)
 
     //when
-    blockBroadcast.broadcastNewBlocks(Seq(firstBlock, secondBlock), Map(peer -> initialPeerInfo))
+    blockBroadcast.broadcastBlocks(Seq(firstBlock, secondBlock), Map(peer -> initialPeerInfo))
 
     //then
     etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(firstBlock, peer.id))
+    etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(firstBlockNewHashes, peer.id))
+    etcPeerManagerProbe.expectNoMsg()
+  }
+
+  it should "send block hashes to all peers while the blocks only to sqrt of them" in new TestSetup {
+    //given
+    val firstHeader: BlockHeader = baseBlockHeader.copy(number = initialPeerInfo.maxBlockNumber + 4)
+    val firstBlockNewHashes = NewBlockHashes(Seq(PV62.BlockHash(firstHeader.hash, firstHeader.number)))
+    val firstBlock = NewBlock(Block(firstHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty - 2)
+
+    val peer2Probe = TestProbe()
+    val peer2 = Peer(new InetSocketAddress("127.0.0.1", 0), peer2Probe.ref, false)
+    val peer3Probe = TestProbe()
+    val peer3 = Peer(new InetSocketAddress("127.0.0.1", 0), peer3Probe.ref, false)
+    val peer4Probe = TestProbe()
+    val peer4 = Peer(new InetSocketAddress("127.0.0.1", 0), peer4Probe.ref, false)
+
+    //when
+    val peersWithInfo = Map(
+      peer -> initialPeerInfo,
+      peer2 -> initialPeerInfo,
+      peer3 -> initialPeerInfo,
+      peer4 -> initialPeerInfo)
+    val blockBroadcastFirstTwoPeers = new BlockBroadcast {
+      override val etcPeerManager: ActorRef = etcPeerManagerProbe.ref
+
+      override def obtainRandomPeerSubset(peers: Set[Peer]): Set[Peer] = Set(peer, peer2)
```

Comment:
Wouldn't it be better to test the actual implementations? Just check that there are 2 receiving peers within a subset of `peersWithInfo`. `etcPeerManagerProbe.expectMsgPF` may come handy.